pkg/cloudprovider/aws/amifamily/ami.go

```diff
@@ -46,3 +83,35 @@ func (p *AMIProvider) Get(ctx context.Context, instanceType cloudprovider.Instan
 	logging.FromContext(ctx).Debugf("Discovered %s for query %q", ami, ssmQuery)
 	return ami, nil
 }
+
+func (p *AMIProvider) GetAMIRequirements(ctx context.Context, providerRef *v1alpha5.ProviderRef) (map[string]scheduling.Requirements, error) {
+	amiRequirements := make(map[string]scheduling.Requirements)
+	if providerRef != nil {
+		var ant v1alpha1.AWSNodeTemplate
+		if err := p.kubeClient.Get(ctx, types.NamespacedName{Name: providerRef.Name}, &ant); err != nil {
+			return amiRequirements, fmt.Errorf("retrieving provider reference, %w", err)
+		}
+		for _, ami := range ant.Spec.AMIs {
+			amiRequirements[ami.ID] = scheduling.NewAMIRequirements(ami)
+		}
+	}
+	return amiRequirements, nil
+}
+
+func getAMIOverride(instanceType cloudprovider.InstanceType, amiRequirements map[string]scheduling.Requirements) string {
+	for amiID, requirements := range amiRequirements {
+		if err := instanceType.Requirements().Compatible(requirements); err == nil {
+			return amiID
+		}
+	}
+	return ""
+}
+
+// Keys returns a slice of all the keys in a map
+func Keys(m map[string]scheduling.Requirements) []string {
```

Comment:
same here on not exporting?

Reply:
This is just the `lo.Keys` function I think.